pkg/resourcecreator/serviceaccount.go

```diff
@@ -1,17 +1,25 @@
 package resourcecreator
 
 import (
+	"fmt"
+
 	nais "github.com/nais/naiserator/pkg/apis/nais.io/v1alpha1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func ServiceAccount(app *nais.Application) *corev1.ServiceAccount {
+func ServiceAccount(app *nais.Application, projectId string) *corev1.ServiceAccount {
```

Comment:
Put `projectId` into `resourceoptions` and pass that as an argument instead. There are examples of this in i.e. `Deployment()`.